internal/security/setup/option/cache_test.go

```diff
@@ -158,51 +161,39 @@ func setupCacheTest(t *testing.T) func() {
 
 	_ = setup.Init()
 
-	origScratchDir := pkiInitScratchDir
-	testScratchDir, tempDirErr := ioutil.TempDir(curDir, "scratch")
-	if tempDirErr != nil {
-		t.Fatalf("cannot create temporary scratch directory for the test: %v", tempDirErr)
+	os.Unsetenv(envXdgRuntimeDir) // unset env var, so it uses the config toml
+
+	testWorkDir := getWorkDir()
+	testScratchDir := filepath.Join(testWorkDir, pkiInitScratchDir)
+	if err := createDirectoryIfNotExists(testScratchDir); err != nil {
+		t.Fatalf("cannot create resource dir %s for the test: %v", testScratchDir, err)
+	}
+
+	testGeneratedDir := filepath.Join(testWorkDir, pkiInitGeneratedDir)
+	if err := createDirectoryIfNotExists(testGeneratedDir); err != nil {
+		t.Fatalf("cannot create resource dir %s for the test: %v", testGeneratedDir, err)
```

Comment:
change the error message to match the real: it should be `cannot create generated dir`